src/asm/output.c

```diff
@@ -0,0 +1,83 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   output.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: asarandi <asarandi@student.42.us.org>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/02/11 20:52:17 by asarandi          #+#    #+#             */
+/*   Updated: 2018/02/28 01:31:21 by asarandi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "asm.h"
+
+int		update_acb(int acb, int index, int op_type)
+{
+	int	tmp;
+	int shift;
+
+	tmp = 0;
+	shift = 0;
+	if (op_type == T_REG)
+		tmp = REG_CODE;
+	if (op_type == T_DIR)
+		tmp = DIR_CODE;
+	if (op_type == T_IND)
+		tmp = IND_CODE;
+	if (index == 0)
+		shift = 6;
+	if (index == 1)
+		shift = 4;
+	if (index == 2)
+		shift = 2;
+	tmp = (tmp << shift);
+	return (acb + tmp);
+}
+
+void	output_write_data(t_asm *a, t_instruction *ptr, int *i, int j)
+{
+	if (ptr->op_types[j] == T_REG)
+		a->body[(*i)++] = (char)ptr->op_values[j];
```

Comment:
How does this ensure that a->body has enough space? I don't see any resizing allocations in the code.

Reply:
it does not, the buffer is static so in theory if the input is very large there is a chance of data corruption/overwriting other values, however in practice we have the limitation of MAX_CHAMP_SIZE, the assembler will display a warning if the code exceeds that value

Reply:
That looks like a problem.  I'll have a look at this a bit more closely to figure out the underlying problem.